base/broadcast.jl

```diff
@@ -269,22 +269,14 @@ end
 @inline broadcast_elwise_op(f, As...) =
     broadcast!(f, similar(Array{promote_eltype_op(f, As...)}, broadcast_indices(As...)), As...)
 
-ftype(f, A) = typeof(f)
-ftype(f, A...) = typeof(a -> f(a...))
-ftype(T::Type, A...) = Type{T}
-typestuple(a) = (Base.@_pure_meta; Tuple{eltype(a)})
-typestuple(T::Type) = (Base.@_pure_meta; Tuple{Type{T}})
-typestuple(a, b...) = (Base.@_pure_meta; Tuple{typestuple(a).types..., typestuple(b...).types...})
-ziptype(A) = typestuple(A)
-ziptype(A, B) = (Base.@_pure_meta; Iterators.Zip2{typestuple(A), typestuple(B)})
-@inline ziptype(A, B, C, D...) = Iterators.Zip{typestuple(A), ziptype(B, C, D...)}
-
-_broadcast_type(f, T::Type, As...) = Base._return_type(f, typestuple(T, As...))
-_broadcast_type(f, A, Bs...) = Base._default_eltype(Base.Generator{ziptype(A, Bs...), ftype(f, A, Bs...)})
+eltypestuple(a) = (Base.@_pure_meta; Tuple{eltype(a)})
```

Comment:
Sorry for commenting above without providing a context. I was thinking that we could change the `promote_op` methods to take a function and the tuple of the `eltype`s and remove the `promote_op(Any...) = Any`, so it would make sense having `eltypestuple` somewhere in `Base`